### PR TITLE
An example of generating the bathymetry on a Warped grid

### PR DESCRIPTION
**Pull request by simone-silvestri**
This is the output of `examples/generate_bathymetry.jl`

Screenshot 2023-12-04 at 10 42 39 PM

### CODE REVIEW

**Comment by simone-silvestri** [comment]
Another example at 1/20th of a degree (quite similar actually by eye)

Screenshot 2023-12-04 at 10 42 39 PM

**Comment by glwagner** [comment]
This looks great! The name "warped" seems nice and the API seems interpretable to me. Is `OrthogonalSphericalShellGrids` a registered package? I think that will help with versioning if it is.

File: src/Bathymetry.jl [hunk 1]
```diff
@@ -22,14 +29,38 @@ using Printf
...
+- dir: directory of the bathymetry-containing file
+
+- filename: file containing bathymetric data. Must be netcdf with fields:
+            (1) `lat` vector of latitude nodes
+            (2) `lon` vector of longitude nodes
+            (3) `z` matrix of depth values
+
+- interpolation_passes: regridding/interpolation passes. The bathymetry is interpolated in
+                        `interpolation_passes - 1` intermediate steps. With more steps the 
+                        final bathymetry will be smoother.
```
Comment (glwagner) [inline src/Bathymetry.jl:54]: Can you add more to this description? I understand that it smooths things out but I don't understand how it works.

File: src/Bathymetry.jl [hunk 2]
```diff
@@ -22,14 +29,38 @@ using Printf
...
+
 """
 function regrid_bathymetry(target_grid;
                            height_above_water = nothing,
                            minimum_depth = 0,
                            dir = joinpath(@__DIR__, "..", "data"),
                            url = "https://www.ngdc.noaa.gov/thredds/fileServer/global/ETOPO2022/60s/60s_surface_elev_netcdf", 
-                           filename = "ETOPO_2022_v1_60s_N90W180_surface.nc")
+                           filename = "ETOPO_2022_v1_60s_N90W180_surface.nc",
+                           interpolation_passes = 10)
```
Comment (glwagner) [inline src/Bathymetry.jl:63]:
```suggestion
                           interpolation_passes = 0)
```

10 seems random

File: src/DataWrangling/ECCO2.jl [hunk 3]
```diff
@@ -1,19 +1,50 @@
...
+using KernelAbstractions: @kernel, @index
 using NCDatasets
 
-temperature_filename = "THETA.1440x720x50.19920102.nc"
-salinity_filename = "SALT.1440x720x50.19920102.nc"
-effective_ice_thickness_filename = "SIheff.1440x720.19920102.nc"
+import Oceananigans.Fields: set!
+
+# Ecco field used to set model's initial conditions
+struct ECCO2Data
```
Comment (glwagner) [inline src/DataWrangling/ECCO2.jl:18]:
```suggestion
struct ECCO2Metadata
```

this isn't the data itself, but not sure what to call it

File: Project.toml [hunk 4]
```diff
@@ -9,10 +9,12 @@ CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
 CubicSplines = "9c784101-8907-5a6d-9be6-98f00873c89b"
 DataDeps = "124859b0-ceae-595e-8997-d05f6a7a8dfe"
 Downloads = "f43a241f-c20a-4ad4-852c-f6b1247861c6"
+FFMPEG = "c87230d0-a227-11e9-1b43-d7ebe4e7570a"
```
Comment (navidcy) [inline Project.toml:12]: why we need this?

**Comment by simone-silvestri** [comment]
This PR is Stale

**Pull request closed without merging**